Library/Homebrew/command.rb

```diff
@@ -0,0 +1,157 @@
+module Homebrew
+  class Command
+    attr_reader :command_name, :valid_options, :description, :help_output,
+      :man_output
+
+    def initialize_variables
+      @valid_options = []
+      @root_options = []
+      @argv = ARGV.dup
+    end
+
+    def options(&block)
+      initialize_variables
+      @parent = nil
+      instance_eval(&block)
+      generate_help_and_manpage_output
+    end
+
+    def cmd_name(cmd)
+      @command_name = cmd
+    end
+
+    def add_valid_option(option_hash)
+      @valid_options.push(option_hash)
+    end
+
+    def option(*args, **option_hash, &block)
+      option_hash[:option] = "--#{args[0]}"
```

Comment:
Sorry, I still don't understand. Can you explain what this line is and what a typical array/hash looks like in this case?

---

As a general comment, that I'm finding this code so hard to understand and review means it's a long way off being mergeable, unfortunately.